File: Examples/objective-c/ProximityContent/ProximityContent/Estimote/BeaconDetails.m
//
// Please report any problems with this app template to contact@estimote.com
//

#import "BeaconDetails.h"

@implementation BeaconDetails

- (instancetype)initWithBeaconName:(NSString *)beaconName backgroundColor:(UIColor *)backgroundColor {
    self = [super init];
    if (self) {
        _beaconName = [beaconName copy];
        _backgroundColor = backgroundColor;
    }
    return self;
}

- (instancetype)initWithBeaconName:(NSString *)beaconName beaconColor:(ESTColor)beaconColor {
    return [self initWithBeaconName:beaconName backgroundColor:[self backgroundColorForBeaconColor:beaconColor]];
}

+ (UIColor *)neutralColor {
    return [UIColor colorWithRed: 160/255.0 green: 169/255.0 blue: 172/255.0 alpha: 1.0];
}

- (UIColor *)backgroundColorForBeaconColor:(ESTColor)beaconColor {
    static NSDictionary *backgroundColors = nil;
    if (backgroundColors == nil) {
        backgroundColors =
        [[NSDictionary alloc] initWithObjectsAndKeys:
         [UIColor colorWithRed: 109/255.0 green: 170/255.0 blue: 199/255.0 alpha: 1.0], [NSNumber numberWithInt:ESTColorIcyMarshmallow],
         [UIColor colorWithRed:  36/255.0 green:  24/255.0 blue:  93/255.0 alpha: 1.0], [NSNumber numberWithInt:ESTColorBlueberryPie],
         [UIColor colorWithRed: 155/255.0 green: 186/255.0 blue: 160/255.0 alpha: 1.0], [NSNumber numberWithInt:ESTColorMintCocktail],
         [UIColor colorWithRed:  84/255.0 green:   0/255.0 blue:  61/255.0 alpha: 1.0], [NSNumber numberWithInt:ESTColorSweetBeetroot],
         [UIColor colorWithRed: 195/255.0 green: 192/255.0 blue:  16/255.0 alpha: 1.0], [NSNumber numberWithInt:ESTColorLemonTart],
         [UIColor colorWithRed: 219/255.0 green: 122/255.0 blue: 167/255.0 alpha: 1.0], [NSNumber numberWithInt:ESTColorCandyFloss],
         nil];
    }
    UIColor *backgroundColor = [backgroundColors objectForKey:[NSNumber numberWithInt:beaconColor]];
    if (backgroundColor) {
        return backgroundColor;
    }
    return [BeaconDetails neutralColor];
}

@end

File: Examples/swift/ProximityContent/ProximityContent/Estimote/BeaconDetails.swift
//
// Please report any problems with this app template to contact@estimote.com
//

import UIKit

class BeaconDetails {

    let beaconName: String
    let backgroundColor: UIColor

    init(beaconName: String, backgroundColor: UIColor) {
        self.beaconName = beaconName
        self.backgroundColor = backgroundColor
    }

    convenience init(beaconName: String, beaconColor: ESTColor) {
        self.init(
            beaconName: beaconName,
            backgroundColor: BeaconDetails.backgroundColor(for: beaconColor))
    }

    private static let backgroundColors: [ESTColor: UIColor] = [
        .icyMarshmallow: UIColor(red: 109/255.0, green: 170/255.0, blue: 199/255.0, alpha: 1.0),
        .blueberryPie:   UIColor(red:  36/255.0, green:  24/255.0, blue:  93/255.0, alpha: 1.0),
        .mintCocktail:   UIColor(red: 155/255.0, green: 186/255.0, blue: 160/255.0, alpha: 1.0),
        .sweetBeetroot:  UIColor(red:  84/255.0, green:   0/255.0, blue:  61/255.0, alpha: 1.0),
        .lemonTart:      UIColor(red: 195/255.0, green: 192/255.0, blue:  16/255.0, alpha: 1.0),
        .candyFloss:     UIColor(red: 219/255.0, green: 122/255.0, blue: 167/255.0, alpha: 1.0)
    ]

    static let neutralColor = UIColor(red: 160/255.0, green: 169/255.0, blue: 172/255.0, alpha: 1.0)

    private static func backgroundColor(for beaconColor: ESTColor) -> UIColor {
        return backgroundColors[beaconColor] ?? neutralColor
    }

}
